### ISSUE TITLE
fluentbit crashes with SIGSEGV error when nested incorrect json  in the record

### ISSUE THREAD

**Issue by imulot** [fluent/fluent-bit#2957]
## Bug Report

**Describe the bug**
fluentbit crashes with SIGSEGV error when nested incorrect json is in the record

**To Reproduce**

- Add the following record :

```
{"@version":1,"@timestamp":"2021-01-18T16:35:41.965+01:00","sequence":645,"level":"ERROR","message":"{\"foo\":\"\",\"bar\":\"test\"}{\"foo\":\"\",\"bar\":\"test\"}","threadName":"Thread-17"}
```
- Configuration: 

```
    [PARSER]
        Name             kube_parser
        Format           json
        Time_Key         @timestamp
        Time_Format      %Y-%m-%dT%H:%M:%S.%L %z
        Time_Keep        On
        Decode_Field_As  json     message  try_next
        Decode_Field_As  escaped  message
```
	
- Stack Trace:

```
		[engine] caught signal (SIGSEGV)
		#0  0x55afdec580ce      in  ???() at ???:0
		#1  0x55afdec3ea46      in  ???() at ???:0
		#2  0x55afdeb64e91      in  lua_arraylength() at plugins/filter_lua/lua.c:110
		#3  0x55afdeb65205      in  lua_tomsgpack() at plugins/filter_lua/lua.c:213
		#4  0x55afdeb65344      in  lua_tomsgpack() at plugins/filter_lua/lua.c:241
		#5  0x55afdeb65b36      in  cb_lua_filter() at plugins/filter_lua/lua.c:466
		#6  0x55afdeab5a66      in  flb_filter_do() at src/flb_filter.c:118
		#7  0x55afdeadb8a8      in  flb_input_chunk_append_raw() at src/flb_input_chunk.c:473
		#8  0x55afdeaf4b76      in  process_content() at plugins/in_tail/tail_file.c:339
		#9  0x55afdeaf656c      in  flb_tail_file_chunk() at plugins/in_tail/tail_file.c:877
		#10 0x55afdeaf1286      in  in_tail_collect_static() at plugins/in_tail/tail.c:140
		#11 0x55afdeab551c      in  flb_input_collector_fd() at src/flb_input.c:979
		#12 0x55afdeac1db3      in  flb_engine_handle_event() at src/flb_engine.c:307
		#13 0x55afdeac1db3      in  flb_engine_start() at src/flb_engine.c:559
		#14 0x55afdea362f4      in  flb_main() at src/fluent-bit.c:1035
		#15 0x55afdea36342      in  main() at src/fluent-bit.c:1048
		#16 0x7fb0cf25109a      in  ???() at ???:0
		#17 0x55afdea33fd9      in  ???() at ???:0
		#18 0xffffffffffffffff  in  ???() at ???:0	
```

**Expected behavior**

- fluent-bit must not crash with SIGSEGV with incorrect formatted JSON
```
"message":"{\"foo\":\"\",\"bar\":\"test\"}{\"foo\":\"\",\"bar\":\"test\"}"
```
	
- After fixed, JSON no issue.
```
"message":"[{\"foo\":\"\",\"bar\":\"test\"},{\"foo\":\"\",\"bar\":\"test\"}]"
```

**Your Environment**
* Version used: Reproduced on 1.3, 1.4.5, 1.5.7
* Configuration: default GKE config
* Environment name and version: Openshift v3.11
* Server type and version: x86
* Operating System and version: RHEL7
* Filters and plugins: see above

**Comment by imulot** [comment]
Seems related to jsmn.

Parser seems not detected that JSON is incorrectly formatted.
Decode_Field_As json     message  try_next

For example if I add string at the beginning of message field, fluentbit doesn't crash.

`"message":"FOO:{\"foo\":\"\",\"bar\":\"test\"}{\"foo\":\"\",\"bar\":\"test\"}","threadName":"Thread-17"}`

**Comment by edsiper** [comment]
thanks for reporting this issue. A fix has been pushed to GIT master and also will be part of v1.6 release (this Friday).

fix: bd43f031

**Comment by imulot** [comment]
@edsiper,
Are you sure it will be backported to v1.6 release ?
flb_pack_json_recs() used by the fix is not present in 1.6.10 release.

I attempt to build (with docker build)  1.7-rc3 to test the fix .

Build failed : 

```
STEP 20: COPY --from=builder /usr/lib/x86_64-linux-gnu/libz* /usr/lib/x86_64-linux-gnu/
Error: error building at STEP "COPY --from=builder /usr/lib/x86_64-linux-gnu/libz* /usr/lib/x86_64-linux-gnu/": error evaluating symlinks "/var/lib/containers/storage/vfs/dir/e1a28205d367980ba4faf262ee61a990ab77c0c8b200a3950a67c473f1a20d66/usr/lib/x86_64-linux-gnu/libz.so": lstat /usr/lib/x86_64-linux-gnu: no such file or directory
```
